solr/solrj/src/java/org/apache/solr/client/solrj/impl/CloudSolrClient.java

```diff
@@ -1300,9 +1306,19 @@ private CompletableFuture<DocCollection> triggerCollectionRefresh(String collect
               future.completeExceptionally(t);
             }
           } else {
```

Comment:
You're trying hard to load a collection's state in spite of a strong indication that our CloudSolrClient is shutting down.  Why bother?  Imagine instead simply assume threadPool is never null and proceed, leading to a RejectedExecutionException (reasonable).  Our close method needn't set threadPool to null.